src/server/http/__tests__/xsrf.js

```diff
@@ -29,10 +30,23 @@ describe('xsrf request filter', function () {
 
     await kbnServer.ready();
 
-    const routeMethods = nonDestructiveMethods.filter(method => method !== 'HEAD').concat(destructiveMethods);
     kbnServer.server.route({
       path: '/xsrf/test/route',
-      method: routeMethods,
+      method: nonDestructiveMethods.filter(method => method !== 'HEAD'),
+      handler: function (req, reply) {
+        reply(null, 'ok');
+      }
+    });
+
+    kbnServer.server.route({
+      path: '/xsrf/test/route',
+      method: destructiveMethods,
+      config: {
+        // Disable payload parsing to make HapiJS server accept any content-type header.
```

Comment:
Yeah, I might have been over-thinking the "production setup" vs "test setup" thingy.

Regarding testing: I played around with this and did these two requests. The first specifies `kbn-xsrf`, but an invalid `Content-Type`, while the second removes the `kbn-xsrf`.

```
$ curl \
-X POST \
-d '{ "attributes": { "title": "Test pattern" } }' \
-H "Content-Type: text/plain" \
-H "kbn-xsrf: 6.0.0" \
http://localhost:5601/api/saved_objects/index-pattern
{"type":"index-pattern","id":"AVxVAEEIK58oVmHlSYBw","version":1,"attributes":{"title":"Test pattern"}}
```

then

```
~/dev/elastic/kibana azasypkin-issue-11821-verify-content-type*
$ curl \
> -X POST \
> -d '{ "attributes": { "title": "Test pattern" } }' \
> -H "Content-Type: text/plain" \
> http://localhost:5601/api/saved_objects/index-pattern
{"statusCode":400,"error":"Bad Request","message":"Request must contain a content-type header of either application/json or application/x-ndjson. The content-type header for current request is text/plain."}
```

I'm not really sure what we should do here, but when we decide (cc @epixa @spalger) I think something like this should be covered in the tests to make the intention clear for later.